packages/components/src/internal/components/samples/actions.ts

```diff
@@ -370,3 +374,107 @@ export function getDeleteSharedSampleTypeUrl(typeId: number): string {
         returnUrl: window.location.pathname + '#/samples',
     }).toString();
 }
+
+function getSamplesIdsNotFound(queryName: string, orderedIds: string[]) : Promise<string[]> {
+    return new Promise( (resolve, reject) => {
+        Query.selectDistinctRows({
+                schemaName: SCHEMAS.EXP_TABLES.SCHEMA,
+                queryName,
+                column: 'Ordinal',
+                sort: 'Ordinal',
+                success: result => {
+                    const ordinals = result.values
+                    let index = 0;
+                    let oIndex = 0;
+                    let missingIds = [];
+                    // find the gaps in the ordinals values as these correspond to ids we could not find
+                    while (index < orderedIds.length) {
+                        if (oIndex >= ordinals.length || ordinals[oIndex] !== index + 1) {
+                            missingIds.push(orderedIds[index])
+                        } else {
+                            oIndex++;
+                        }
+                        index++;
+                    }
+                    resolve(missingIds);
+                },
+                failure: reason => {
+                    console.error("There was a problem determining the missing Ids", reason);
+                    reject(reason);
+                }
+            });
+    });
+}
+
+export function getFindSamplesByIdData(staleQueryName?: string) : Promise<{ queryName: string, missingIds?: { [key: string]: string[] } }> {
```

Comment:
staleQueryName? what is stale about it? Also, I don't see this param used anywhere, can it be removed?